walletlibrary/src/test/java/com/microsoft/walletlibrary/networking/entities/openid4vci/CredentialMetadataTest.kt

```diff
@@ -0,0 +1,187 @@
+        assertThat(actualException?.message).isEqualTo("Authorization server $authorizationServer1 not found in Credential Metadata.")
+        assertThat((actualException as OpenId4VciValidationException).code).isEqualTo(VerifiedIdExceptions.MALFORMED_CREDENTIAL_METADATA_EXCEPTION.value)
+    }
+
+    @Test
+    fun validateAuthorizationServers_NoAuthorizationServerInMetadata_ThrowsException() {
+        // Arrange
+        val credentialMetadata = CredentialMetadata()
+
+        // Act
+        val actualResult = runCatching {
+            credentialMetadata.validateAuthorizationServers(credentialOffer)
+        }
+
+        // Assert
+        assertThat(actualResult.isSuccess).isFalse
+        val actualException = actualResult.exceptionOrNull()
+        assertThat(actualException).isInstanceOf(OpenId4VciValidationException::class.java)
+        assertThat(actualException?.message).isEqualTo("Authorization servers property missing in credential metadata.")
+        assertThat((actualException as OpenId4VciValidationException).code).isEqualTo(VerifiedIdExceptions.INVALID_PROPERTY_EXCEPTION.value)
+    }
+
+    @Test
+    fun verifyIfCredentialIssuerAndSignedMetadataExist_NoCredentialIssuer_ThrowsException() {
+        // Arrange
+        val credentialMetadata = CredentialMetadata()
+
+        // Act
+        val actualResult = runCatching {
+            credentialMetadata.verifyIfCredentialIssuerAndSignedMetadataExist()
+        }
+
+        // Assert
+        assertThat(actualResult.isSuccess).isFalse
+        val actualException = actualResult.exceptionOrNull()
+        assertThat(actualException).isInstanceOf(OpenId4VciValidationException::class.java)
+        assertThat(actualException?.message).isEqualTo("Credential metadata does not contain credential_issuer.")
+        assertThat((actualException as OpenId4VciValidationException).code).isEqualTo(VerifiedIdExceptions.MALFORMED_CREDENTIAL_METADATA_EXCEPTION.value)
+    }
+
+    @Test
+    fun verifyIfCredentialIssuerAndSignedMetadataExist_NoSignedMetadata_ThrowsException() {
+        // Arrange
+        val credentialMetadata = CredentialMetadata(credential_issuer = "credential_issuer")
+
+        // Act
+        val actualResult = runCatching {
+            credentialMetadata.verifyIfCredentialIssuerAndSignedMetadataExist()
+        }
+
+        // Assert
+        assertThat(actualResult.isSuccess).isFalse
+        val actualException = actualResult.exceptionOrNull()
+        assertThat(actualException).isInstanceOf(OpenId4VciValidationException::class.java)
+        assertThat(actualException?.message).isEqualTo("Credential metadata does not contain signed_metadata.")
+        assertThat((actualException as OpenId4VciValidationException).code).isEqualTo(VerifiedIdExceptions.MALFORMED_CREDENTIAL_METADATA_EXCEPTION.value)
+    }
+
+    @Test
+    fun transformIssuerStyle_NoDisplayDefinition_ReturnsRequesterStyleWithNoIssuerName() {
```

Comment:
Will you add a test for successfully transformingIssuerStyle?

Reply:
I have actually removed the test for transform from this PR. I will add the unit tests for transforming both display for VC and display for issuer in a later PR. Mocking the Android class will take additional time so don't want to block this PR. This method is not used in this PR yet.